### PR TITLE
Feature/profile

### PR DESCRIPTION
**Pull request by Adventador95**
Modified the UserProfile page to show the reward points the logged in user has available with a tooltip explaining the details. Also made sure that new users get a record in the database when they create an account (Google also).

### CODE REVIEW

File: frontend/src/app/register/register.component.ts [hunk 1]
```diff
@@ -50,10 +51,17 @@ export class RegisterComponent implements OnInit {
       securityAnswer: this.securityAnswerControl.value
     }
 
+    const reward = { 
+      amount: 0 // Default value, new accounts have 0 bonus points
```
Comment (elyst) [inline frontend/src/app/register/register.component.ts]: delete comment.

File: frontend/src/app/register/register.component.ts [hunk 2]
```diff
@@ -50,10 +51,17 @@ export class RegisterComponent implements OnInit {
...
+    }
+
     this.userService.save(user).subscribe((response: any) => {
       this.securityAnswerService.save({UserId: response.id, answer: this.securityAnswerControl.value,
         SecurityQuestionId: this.securityQuestionControl.value}).subscribe(() => {
-          this.router.navigate(['/login'])
+          // Reward stuff here
```
Comment (elyst) [inline frontend/src/app/register/register.component.ts]: delete comment.

File: routes/userProfile.js [hunk 3]
```diff
@@ -13,29 +15,37 @@ module.exports = function getUserProfile () {
...
         models.User.findByPk(loggedInUser.data.id).then(user => {
-          let jadeTemplate = buf.toString()
-          let username = user.dataValues.username
-          if (username.match(/#\{(.*)\}/) !== null && !utils.disableOnContainerEnv()) {
-            req.app.locals.abused_ssti_bug = true
-            const code = username.substring(2, username.length - 1)
-            try {
-              eval(code) // eslint-disable-line no-eval
-            } catch (err) {
+          models.Reward.findOne({ where: { userId: loggedInUser.data.id } }).then(bonus => { // Find reward model for user
```
Comment (elyst) [inline routes/userProfile.js]: delete comment.

File: frontend/src/app/register/register.component.ts (hunk 2 above)
Comment (jurrewolff) [inline frontend/src/app/register/register.component.ts]: Comments need to go.

**Pull request merged**